scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -299,6 +299,49 @@ object Defn {
                     templ: Template) extends Defn with Member.Term {
     checkFields(templ.is[Template.Quasi] || templ.stats.forall(!_.is[Ctor]))
   }
+  @ast class Enum(mods: List[Mod],
+                  name: scala.meta.Type.Name,
+                  tparams: List[scala.meta.Type.Param],
+                  ctor: Ctor.Primary,
+                  templ: Template) extends Defn with Member.Type
+
+  object Enum {
+    @ast class Name(value: Predef.String @nonEmpty)
+      extends scala.meta.Name with Term.Ref //with Member.Term
+    // example:
+    // enum Try {
+    //   case Success(value: String)
+    //   case Failure(e: Throwable)
+    // }
+    @ast class Case(mods: List[Mod],
+                    name: Term.Name /* not Type.Name because only companion is visible */,
+                    tparams: List[scala.meta.Type.Param],
+                    ctor: Ctor.Primary,
+                    inits: List[Init]) extends Defn with Member.Term {
+      //      checkFields(
+      //        parent.isEmpty || {
+      //          val p = parent.get
+      //          p.is[Template] && (p.parent.isEmpty || p.parent.get.is[Defn.Enum])
+      //        }
+      //      )
+      checkParent(ParentChecks.CaseEnum)
+
+    }
+    // example:
+    // enum Color {
+    //   case Red, Blue, Green
+    // }
+    @ast class RepeatedCase(mods: List[Mod], cases: List[Enum.Name]) extends Defn {
+      //      checkFields(
+      //        parent.isEmpty || {
+      //          val p = parent.get
+      //          p.is[Template] && (p.parent.isEmpty || p.parent.get.is[Defn.Enum])
+      //        }
+      //      )
+      checkParent(ParentChecks.CaseEnum)
```

Comment:
This check is written for `Enum.Case` when it should be for `Case.RepeatedCase`

```
[error] /home/travis/build/scalameta/trees/scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala:341:32: type mismatch;
[error]  found   : (scala.meta.Defn.Enum.Case, scala.meta.Tree, String) => Boolean
[error]  required: (scala.meta.Defn.Enum.RepeatedCase.DefnEnumRepeatedCaseImpl, scala.meta.Tree, String) => Boolean
[error]       checkParent(ParentChecks.CaseEnum)
[error]                                ^
[error] one error found
```

You can try to broaden the first argument in
```diff
- def CaseEnum(tree: Defn.Enum.Case, parent: Tree, destination: String) : Boolean= {
+ def CaseEnum(tree: Defn, parent: Tree, destination: String) : Boolean= {
```